### DISCUSSION TITLE
Building a Custom Retriever Model like CoLBERT for DSPy

### DISCUSSION THREAD

**Discussion by brando90** [stanfordnlp/dspy#1019] [Q&A]
I want to make my own retriever model. I have either a connection to a database or a path to a file with lots of JSON files, or it's already been loaded as a list of dictionaries (or what is the best format for this to work for DSPy?), and **I want to be able to make a retriever model using DSPy, just like CoLBERT does**. How does one do that? 

Does somebody have an example or a tutorial or sample code on what the code is to build a CoLBERt like RM for DSPy? Thank you!

**Comment by hieunv3vmo** [comment]
You can checkout the document here https://dspy-docs.vercel.app/docs/deep-dive/retrieval_models_clients/custom-rm-client

**Comment by techsackk** [comment]
@brando90 Have you implemented this custom retriever that points to the JSON files? If so, can you please share the code here?
